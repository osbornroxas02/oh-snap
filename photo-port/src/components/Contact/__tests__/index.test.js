import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '..';

afterEach(cleanup)

describe('ContactForm component', () => {
    it('renders', () => {
        render(<Contact />);
    });


    /*
        In the next test, use the data-testid attribute for the <h1> tag in the ContactForm component to match the text content to be "Contact me".
    */

    // Second Test
    it('matches text content', () => {
        const { asFragment } = render(<contact-form currentCategory={Contact} />)
        expect(asFragment()).toMatchSnapshot()
    });

    /*
        The last test should also use the data-testid attribute to locate the <button>, to match the text content "Submit".
        
    */
    it('renders', () => {
        const { asFragment } = render(<contact-form currentCategory={Contact} /> )
        expect(asFragment()).toMatchSnapshot()
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<contact-form currentCategory={Contact} />)
        expect(asFragment('button')).toMatchSnapshot()
    })
})