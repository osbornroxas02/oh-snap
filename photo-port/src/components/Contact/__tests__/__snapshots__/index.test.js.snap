// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component matches snapshot 1`] = `
<DocumentFragment>
  <contact-form />
</DocumentFragment>
`;

exports[`ContactForm component matches text content 1`] = `
<DocumentFragment>
  <contact-form />
</DocumentFragment>
`;

exports[`ContactForm component renders 1`] = `
<DocumentFragment>
  <contact-form />
</DocumentFragment>
`;
